### ISSUE TITLE
Review "Party"

### ISSUE THREAD

**Issue by yfarjoun** [samtools/htsjdk#548]
Tomorrow, April 5th, we will make a concerted effort to review the oldest 5 PR's. Folks should feel free to join us at:

https://plus.google.com/hangouts/_/broadinstitute.com/htsjdk-review?authuser=0&hceid=ZmFyam91bkBicm9hZGluc3RpdHV0ZS5vcmc.itolfci2q1kupsqd5mp2q5ag90

The PR's that will be reviewed are:

https://github.com/samtools/htsjdk/pull/232
https://github.com/samtools/htsjdk/pull/313
https://github.com/samtools/htsjdk/pull/353
https://github.com/samtools/htsjdk/pull/358
https://github.com/samtools/htsjdk/pull/361

If you cannot join but would like to comment about any of these PR's, please do so soon.

**Comment by nh13** [comment]
#232 I am so excited!  Thank-you @yfarjoun and others for organizing this.  I wont be able to attend (Bio-IT) but I have tried to comment on the other PRs.

**Comment by yfarjoun** [comment]
Summary of action-items/descitions:

- https://github.com/samtools/htsjdk/pull/232 trivial review requests, please merge after bugfix for issue ???
- https://github.com/samtools/htsjdk/pull/313 back to OC few remaining comments and a rebase
- https://github.com/samtools/htsjdk/pull/353 request for failing test-case and a rebase to fix commits
- https://github.com/samtools/htsjdk/pull/358 looks good, will be reviewed by @vadimzalunin 
- https://github.com/samtools/htsjdk/pull/361 closed due to lack of interest

**Comment by yfarjoun** [comment]
Reopening this to keep history.

On April 19 we will be following up on PR's from last time and looking at 

https://github.com/samtools/htsjdk/pull/383
https://github.com/samtools/htsjdk/pull/408
https://github.com/samtools/htsjdk/pull/421
https://github.com/samtools/htsjdk/pull/443
https://github.com/samtools/htsjdk/pull/449

**Comment by droazen** [comment]
Outcome of today's code review party:

https://github.com/samtools/htsjdk/pull/232 Waiting on a final review from @jmarshall 
https://github.com/samtools/htsjdk/pull/313 Merged!
https://github.com/samtools/htsjdk/pull/353 Closed due to no reply from author
https://github.com/samtools/htsjdk/pull/358 Closed due to no reply from author, but @vadimzalunin has agreed to incorporate the same fix into an upcoming branch, with a test.

https://github.com/samtools/htsjdk/pull/383 Decided on a way forward, added a comment with a roadmap
https://github.com/samtools/htsjdk/pull/408 Will wait 2 weeks for a response to see if there's still interest
https://github.com/samtools/htsjdk/pull/421 Closed as dupe of 408
https://github.com/samtools/htsjdk/pull/443 Agreed on changes to be made, @droazen will review before next meeting
https://github.com/samtools/htsjdk/pull/449 Punted until next meeting

**Comment by yfarjoun** [comment]
On May 3rd we will be following up on PR's from last time and looking at

https://github.com/samtools/htsjdk/pull/458
https://github.com/samtools/htsjdk/pull/478
https://github.com/samtools/htsjdk/pull/480
https://github.com/samtools/htsjdk/pull/489
https://github.com/samtools/htsjdk/pull/495

**Comment by yfarjoun** [comment]
Outcome of today's code review party:

https://github.com/samtools/htsjdk/pull/232 Waiting on a final-final review from @jmarshall 
https://github.com/samtools/htsjdk/pull/383 Waiting on @lbergelson and @jarey 
https://github.com/samtools/htsjdk/pull/408 reassigned to @yfarjoun for review
https://github.com/samtools/htsjdk/pull/443 waiting for comments to be addressed, should be closed by next meeting
https://github.com/samtools/htsjdk/pull/449 commented that in the lack of spec PR and interface seem too early.
https://github.com/samtools/htsjdk/pull/458 assigned to @droazen 
https://github.com/samtools/htsjdk/pull/478 few comments, then ready to go.
https://github.com/samtools/htsjdk/pull/480 closed due to complexity of PR and non-response from author.
Rest were punted for next time.

**Comment by jmarshall** [comment]
> #232 Waiting on a final-final review from @jmarshall

Would have if you had mentioned it in the party agenda a couple of days ago! 😛 Okay, will look again…

**Comment by yfarjoun** [comment]
good point. I'll start mentioning all the straggling tickets in the
reminder email.

**Comment by yfarjoun** [comment]
We have many "straggling" ticket for this party. So I think we shouldn't add any new ones, and simply do our best to deal with the ones that are open. 

#495 
#489
#478 
#458
#449
#443 
#408 
#383
#232

**Comment by vadimzalunin** [comment]
Sorry, I won't be able to attend today unfortunately.

**Comment by droazen** [comment]
#232 poked nils and jmarshall
#383 merged
#408 not ready, needs revisions and more review
#443 wait for Louis to come back from vacation
#449 wrote note to author suggesting moving this to the specs community
#458 discussed the approach, agreed that the existing version is ok. @droazen needs to do a formal review
#478 still waiting from @vadimzalunin 
#489 @droazen will talk with @cwhelan in person
#495 waiting on a reply from @tfenne

**Comment by yfarjoun** [comment]
In the next HTS JDK PR review Party, we will be following up on what stayed relavent from last time:
#495 
#489
#478
#458
#443 
#232 

And adding a few new ones:
#538 
#542 
#552 
#557 

As there are several PR's that deal with CRAM, I hope that @vadimzalunin will be able to join us....:-)

**Comment by vadimzalunin** [comment]
I'm going to join. 

@yfarjoun it'be nice to have a bit more info on the list:
#495 VariantContextWriterBuilder 
#489 VCF header line parser
#478 CRAM3 by default
#458 Add SeekablePathStream
#443 default methods to Locatable
#232 output the SAM flag

And adding a few new ones:
#538 Implement CRAI writing
#542 variantcontext.* javadoc
#552 CRAM index queries
#557 100% test coverage for FastqRecord

**Comment by yfarjoun** [comment]
Thanks Vadim. sorry for the curtness.

**Comment by droazen** [comment]
#232 progress on review, end in sight
#408 merged!
#443 review complete, waiting on Louis to come back from vacation
#458 review complete, waiting on Tom to come back from vacation
#478 needs rebase, then can be merged
#480 closed/rejected as problematic
#489 needs final review pass from @droazen
#495 waiting on a reply from @tfenne
#538 blocked by #552
#542 @droazen will ask around methods for answers to @ldgauthier's questions
#552 @droazen will do final pass, then @cmnbroad will squash/rebase/merge
#557 We've chimed in with our opinion that the additional test coverage should be decoupled from the breaking API change.

**Comment by yfarjoun** [comment]
for the next review party, we will re-examine our old friends:
#232 Adding the ability to output the SAM flag field in various formats
#443 add default methods to Locatable
#478 write CRAM3 by default
#538 Implement CRAI writing
#542 improve variantcontext.* javadoc
#552 Fixes and tests for CRAM index queries

And add 4 PR's the the list:
#567 Added a skip list implementation for efficient overlap queries 
#572 added read interface and refactoring FastqRecord
#576 Async BAM decompression
#581 Fixing Genotypes generated by GenotypeBuilder

**Comment by droazen** [comment]
#232 merged!
#443 @lbergelson will address comments by next meeting
#458 merged!
#478 merged!
#489 merged!
#495 merged!
#538 @cmnbroad will rebase and merge before next meeting
#542 @droazen will do another review pass before next meeting
#552 merged!
#557 merged!
#567 merged!
#572

**Comment by droazen** [comment]
#443 @lbergelson will review before next meeting
#538 merged!
#542 should be split into documentation PR and non-doc PR, @droazen will review
#572 discussed, raised questions about design, asked for input from the author
#576 @akiezun will do a final review before next meeting
#581 reviewed, needs second pass from @droazen
#593 needs comment from @vadimzalunin

**Comment by yfarjoun** [comment]
For the next meeting we will review the PRs that remained in the previous week:

#443 add default methods to Locatable
#542 improve variantcontext.* javadoc
#572 RawRead interface
#576 Async Bam writing
#581 mutable extendedattributes
#593 throw exception in cram
#638 SRA tests

and also look at the following PRs:
#637 new algorith for samlocus iteration
#644 MD5 mismatch in cram
#646 Fix self-supression problem with AbstractAsyncWriter
#656 a proposed fix for #519

**Comment by akiezun** [comment]
I may not be able to review #576 for this week's party.

**Comment by droazen** [comment]
Outcome of today's review party:

#443 no progress, but @lbergelson will try to get to this one by next meeting
#542 author has removed code changes from branch, @droazen to do another pass on the remaining documentation changes before next meeting
#572 @droazen to reply to author and suggest an approach
#576 @akiezun waiting on request for additional reviewer
#581 did final review, back to author for changes, then good to merge!
#593 waiting on response from @vadimzalunin, @cmnbroad will email him
#616 merged!
#638 will ping author again, if no response by next party @yfarjoun will take over branch
#637 waiting on upload of file from @yfarjoun to replicate performance results
#644 @cmnbroad will email @vadimzalunin and ask him to reply to @yfarjoun's comment
#646 reviewed, back to author for revisions
#656 this was closed by the author without merging

**Comment by droazen** [comment]
#443 
#542 
#572 
#576 
#581 
#593 
#638 
#637 
#644 
#646

**Comment by droazen** [comment]
Outcome of today's review party:

#443 @lbergelson is addressing review comments
#542 Did thorough review, waiting on reply from author
#572 author has addressed comments, needs a new review from @lbergelson 
#576 pinged @jabbarish for a review
#593 pinged @cmnbroad for status
#638 merged!
#637 @yfarjoun will re-run his performance tests
#644 closed in favor of #671, reviewed #671
#646 merged!

**Comment by vadimzalunin** [comment]
any updates?

**Comment by lbergelson** [comment]
No update today, we cancelled the meeting due to people being on vacation and having conflicting meetings.  

It looks like we forgot to make an update last time we did meet.  There was an issue with hangouts that prevented people at the broad from seeing the people not at the broad and I believe we held two parallel meetings. 

@vadimzalunin Did you get the cancellation email today?

**Comment by yfarjoun** [comment]
Nov1

#443 (Locatable stuff) pinging @lbergelson 
#572 needs a new review @lbergelson 
#576 needs response from @d-cameron 
#637 pinged @ZLyanov waiting for response to review.
#678 asked author to delete the constant MAX_GENOTYPE_QUAL at the same time.
#682 @lbergelson  will commandeer PR and get it to the finishline since @jamesemery is in school

**Comment by droazen** [comment]
11/29

#443 "one day"
https://github.com/samtools/htsjdk/pull/572 needs review from @lbergelson 
#576 will merge after one final rebase
#637 author replied, needs response from @yfarjoun 
#678 still waiting on author
#682 needs refactor, problem is deeper than initially thought

**Comment by droazen** [comment]
#720 @vadimzalunin will review
